Lib/typing.py

```diff
@@ -1393,7 +1393,11 @@ def __mro_entries__(self, bases):
                 return ()
             i = bases.index(self)
             for b in bases[i+1:]:
-                if isinstance(b, _BaseGenericAlias) and b is not self:
+                if b is self:
+                    continue
+                if isinstance(b, _BaseGenericAlias):
+                    return ()
+                if Generic in getattr(b, '__mro__', ()):
```

Comment:
I still don't think this is the proper solution here. Exactly which dunder attributes `types.GenericAlias` delegates to the "underlying" class has changed in the past. It could change again in the future. This feels to me like it's a fragile solution that could easily break.

Reply:
I can say even more: I don't yet understand what ideal `__mro__` should look like for all the cases I've added in tests.